packages/experimental/Switch/src/useSwitch.ts

```diff
@@ -45,12 +40,78 @@ export const useSwitch = (props: SwitchProps): SwitchInfo => {
   const onChangeWithAnimation = React.useCallback(
     (e: InteractionEvent, checked?: boolean) => {
       onChange && onChange(e, checked);
-      animateSwitchKnob();
+      if (isMobile) {
+        if (checked) {
+          startTrackBackgroundAnimation(checked, trackBackgroundAnimation);
+          startTrackAnimation(false, animationConfig, animation, checked);
+        } else {
+          startTrackBackgroundAnimation(checked, trackBackgroundAnimation);
+          startTrackAnimation(false, animationConfig, animation, checked);
+        }
+      }
     },
-    [onChange],
+    [onChange, trackBackgroundAnimation, animationConfig, animation],
   );
 
   const [checkedState, toggleCallback] = useAsToggleWithEvent(defaultChecked, checked, onChangeWithAnimation);
+
+  // Function to change background slowly over time when switch is toggled.
+  const startTrackBackgroundAnimation = React.useCallback((checked: boolean, animation: Animated.Value) => {
+    const toValue = checked ? 0 : 1;
+    Animated.timing(animation, {
+      toValue,
+      duration: 500,
+      useNativeDriver: false,
+    }).start();
+  }, []);
+
+  // Function to transform the knob over track in animated way when switch is toggled.
+  const startTrackAnimation = React.useCallback(
+    (onInit = false, animationConfig: IAnimationConfig, animation: Animated.Value, toggled: boolean) => {
+      Animated.timing(animation, {
+        toValue: toggled
+          ? animationConfig.trackWidth - (animationConfig.thumbWidth + animationConfig.thumbMargin * 2)
+          : onInit
+          ? 0
+          : -(animationConfig.trackWidth + animationConfig.thumbWidth),
+        duration: 300,
+        useNativeDriver: false,
+      }).start();
+    },
+    [],
+  );
+
+  const switchAnimationStyles = {
+    // transform over toggled on position to toggled off position and vice versa
+    thumbAnimatedStyle: {
+      transform: [
+        {
+          translateX: animation,
+        },
+      ],
+    },
+    // Interpolate over toggled on color to toggled off color and vice versa
+    trackBackgroundStyle: {
+      backgroundColor: trackBackgroundAnimation.interpolate({
+        inputRange: [0, 1],
+        outputRange: checkedState
+          ? [animationConfig.toggleOnBgColor, animationConfig.toggleOnBgColor]
+          : [animationConfig.toggleOffBgColor, animationConfig.toggleOffBgColor],
+      }),
+    },
+  };
+
+  //Setting the initial position of the know on track when page loads.
+  if (isMobile) {
+    if (checkedState) {
+      startTrackAnimation(false, animationConfig, animation, checkedState);
```

Comment:
It looks like you're calling the animations during the render phase, not in response to a state change like checkedState changing or toggle being called?

Reply:
It probably makes sense to put this in a useEffect hook that depends on checkedState changing instead of running it on every render

Reply:
Moved into useEffect()